File: Core/Domain/Entities/Device.cs
using Cyviz.Core.Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace Cyviz.Core.Domain.Entities
{
    public class Device
    {
        public string Id { get; set; } // PK

        public string Name { get; set; }

        public DeviceType Type { get; set; }

        public DeviceProtocol Protocol { get; set; }

        public List<string> Capabilities { get; set; }

        public DeviceStatus Status { get; set; }

        public DateTime LastSeenUtc { get; set; }

        public string Firmware { get; set; }

        public string Location { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; } // For ETag
    }
}

File: frontend/cyviz/src/types/device.ts
export interface DeviceListDto {
    id: string;
    name: string;
    type: DeviceType;
    status: DeviceStatus;
    location: string;
    lastSeenUtc: string;
}

export interface DeviceDetailDto {
    id: string;
    name: string;
    type: DeviceType;
    protocol: DeviceProtocol;
    capabilities: string[];
    status: DeviceStatus;
    firmware: string;
    location: string;
    lastSeenUtc: string;
    latestTelemetry?: DeviceSnapshotDto | null;
}

export interface DeviceSnapshotDto {
    deviceId: string;
    timestampUtc: string;
    dataJson: string; // JSON string containing telemetry
}

export type DeviceType = "Display" | "Codec" | "Switcher" | "Sensor";

export type DeviceProtocol = "TcpLine" | "HttpJson" | "EdgeSignalR";

export type DeviceStatus = "Online" | "Offline";

export interface DeviceStatusChangedEvent {
  deviceId: string;
  status: "Online" | "Offline";
}
